pkg/util/write.go

```diff
@@ -0,0 +1,28 @@
+package util
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+)
+
+func WriteIfChanged(filename string, data []byte, perm os.FileMode) (bool, error) {
```

Comment:
okay, so if you set perm=0 for a file that doesn't exist, the open syscall succeeds, and it creates a file with all the permission bits turned off. Kinda makes sense to be honest.

Probably the more logical behavior in this scenario is that if perm=0, the file should be opened with perms 0777 (and then umask is applied). If the file already exists then the 0777 would have no effect. I'll fix it.